include/GenSync/Data/InMemContainer.h

```diff
@@ -0,0 +1,213 @@
+            unique_ptr<DataIteratorBase> clone() const override {
+                return unique_ptr<DataIteratorBase>(new InMemIterator(it_));
+            }
+
+        private:
+            /**Internal Iterator Object*/
+            InnerIter it_;
+    };
+
+    /**
+     * Specific in-memory constant iterator. 
+     */
+    class ConstInMemIterator : public DataIteratorBase {
+        public:
+            //Uses the iterator of a list internally
+            using InnerIter = list<shared_ptr<DataObject>>::const_iterator;
+
+            /**
+             * Constructs an iterator with a given iterator.
+             * @param it The existing iterator.
+             */
+            explicit ConstInMemIterator(InnerIter it) : it_(it) {}
+
+             /**
+             * Returns the DataObject pointer the iterator is pointing at.
+             * @return The DataObject pointer the iterator is pointing at.
+             */
+            shared_ptr<DataObject> operator*() const override { return *it_; }
+
+            /**
+             * Increments the iterator up one element in memory.
+             * @return The iterator before the incrementation.
+             */
+            DataIteratorBase& operator++() override { ++it_; return *this; }
+
+            /**
+             * Compares two DataIterators by their implementations.
+             * @return Whether the iterators point to the same memory address or database position.
+             */
+            bool operator==(const DataIteratorBase& other) const override {
+                auto o = dynamic_cast<const ConstInMemIterator*>(&other);
+                return o && (it_ == o->it_);
+            }
+            
+            /**
+             *  Compares two DataIterators by their implementations.
+             *  @return Whether the iterators do not point to the same memory address or database position.
+             */
+            bool operator!=(const DataIteratorBase& other) const override { return !(*this == other); }
+            
+            /**
+             * Returns a pointer to a new iterator that references the same point.
+             * @return A copy of a given iterator.
+             */
+            unique_ptr<DataIteratorBase> clone() const override {
+                return unique_ptr<DataIteratorBase>(new ConstInMemIterator(it_));
+            }
+
+        private:
+            /**Internal Iterator Object*/
+            InnerIter it_;
+        };
+    
+    /**
+     * Constructs a default In Memory Container
+     */
+    InMemContainer() = default;
+
+    /**
+     * Default Destructor. Clears items to free memory.
+     */
+    ~InMemContainer(){clear();}
+    
+    /**
+     * Returns an iterator that points to the beginning of the container.
+     * @return An iterator that points to the beginning of the container.
+     */
+    iterator begin() override{
+        return DataIterator(std::unique_ptr<DataIteratorBase>(new InMemIterator(myData.begin())));
```

Comment:
Consider using make_unique rather than new for exception safety.  For each "new" ... you should know exactly where it will be "delete"d.